### PR TITLE
feat(evolution): Tier 1 evolvable genes (chromosomes A & B), fully wired

### PR DESCRIPTION
**Pull request by csmangum**
## Summary

Implements Tier 1 of the evolution-surface catalog **and** activates every locus end-to-end:

- 27 new real-valued loci added to `DEFAULT_HYPERPARAMETER_GENES` (default chromosome grows from 4 to 31 evolvable genes).
- All 31 loci are now consumed by live runtime code; nothing is declared-but-unread anymore.

## Runtime-effect audit (post-wiring)

Every gene now has a confirmed consumer pipeline.

### Chromosome A — learning / RL hyperparameters

| Gene | Consumer |
|------|----------|
| `learning_rate`, `gamma`, `epsilon_start`, `epsilon_min`, `epsilon_decay`, `memory_size`, `tau`, `batch_size`, `dqn_hidden_size` | `BaseDQNModule.__init__` and Tianshou builders |
| `rl_train_freq`, `per_alpha`, `per_beta_start`, `per_beta_end` | All Tianshou wrappers' replay buffer |
| `target_update_freq` | Tianshou `DQNPolicy(target_update_freq=…)` (explicit kwarg, not filtered) |
| `ensemble_size` | `RandomForestActionSelector.n_estimators` (auto-injected for `random_forest` / `gradient_boost`) |

### Chromosome B — action-policy priors

Two-stage pipeline:

1. **Base weights** — `move/gather/share/attack/reproduce_weight` are bridged into `core.actions[i].weight` by `AgentCore._customize_action_weights` (at construction) and `AgentCore.refresh_action_weights_from_decision_config` (after chromosome re-application). Per-agent gene values win over environment-level `agent_parameters`; default-valued genes fall through unchanged.
2. **State-aware re-weighting** — `LearningAgentBehavior.decide_action` now calls `compute_action_weights` from the new `farm/core/decision/action_weight_policy.py` module, which scales the base vector by the eight multipliers gated by the three thresholds (`move_mult_no_resources`, `gather_mult_low_resources`, `share_mult_{wealthy,poor}`, `attack_mult_{desperate,stable}`, `reproduce_mult_{wealthy,poor}` gated by `attack_starvation_threshold`, `attack_defense_threshold`, `reproduce_resource_threshold`).

The scaled vector is passed to `DecisionModule.decide_action(... action_weights=...)`, which already biases both Q-value exploitation and weighted-random exploration.

## Code changes

- `farm/core/hyperparameter_chromosome.py` — 27 new gene specs in `DEFAULT_HYPERPARAMETER_GENES` plus matching `DEFAULT_GENE_ENCODINGS` (log/linear 8-bit choices).
- `farm/core/agent/core.py`
  - `_customize_action_weights` resolves overrides from `self.config.decision`; default-aware so unaltered configs preserve legacy behaviour.
  - new `refresh_action_weights_from_decision_config()` for live re-application.
  - `_DECISION_ACTION_WEIGHT_FIELDS` map + cached Pydantic-default lookup.
- `farm/core/initial_diversity.py::_apply_chromosome_to_agent` calls the new refresh helper after applying a fresh chromosome.
- `farm/core/decision/decision.py`
  - `_initialize_tianshou_dqn` forwards `int(self.config.target_update_freq)` instead of the hard-coded `500`.
  - new `_initialize_traditional_ml(algorithm_type)` routes `mlp`/`svm`/`random_forest`/`gradient_boost`/`naive_bayes`/`knn` through `AlgorithmRegistry`, with `ensemble_size` auto-injected as `n_estimators` for ensemble-class algorithms.
- `farm/core/decision/algorithms/tianshou.py::DQNWrapper` passes `target_update_freq` and `estimation_step` as explicit `DQNPolicy` constructor kwargs (with a positive-int guard) instead of relying on the `_EXCLUDED_PARAMS` filter.
- `farm/core/decision/action_weight_policy.py` (new) — `ActionStateSignals`, `extract_signals`, `compute_action_weights`. Pure helpers, fully unit-testable without a full agent stack.
- `farm/core/agent/behaviors/learning.py::LearningAgentBehavior.decide_action` builds the weight vector via `compute_action_weights` (state-aware) instead of feeding raw `action.weight` values; falls back gracefully on signal-extraction failure.

## Tests

Added (28 new tests across 3 files; all passing):

- `tests/test_action_weight_policy.py` — 14 cases for the state-aware re-weighter (each multiplier/threshold gate, enabled-mask, all-zero fallback, length-mismatch input validation).
- `tests/test_decision_config_wiring.py` — 7 cases for `target_update_freq` reaching `DQNPolicy._freq` and `ensemble_size` reaching `RandomForestActionSelector.n_estimators` (with `algorithm_params` precedence and non-ensemble-algorithm isolation).
- `tests/test_action_weight_bridging.py` — 7 cases for per-agent gene → `core.actions[i].weight` flow, the legacy `agent_parameters` precedence rule, and the live `refresh_action_weights_from_decision_config()` path.

Updated (preserving existing semantics under the larger default chromosome):

- `tests/test_hyperparameter_chromosome.py` — derive vector length from `chromosome.evolvable_gene_names()`; rewrite `TestComputeBoundaryPenalty` cases as single-gene chromosomes.
- `tests/runners/test_evolution_experiment.py::test_boundary_penalty_adjusts_candidate_fitness` — compute expected penalty programmatically.
- `tests/analysis/test_speciation.py::_make_fake_chromosome` — explicit 2-gene chromosome so GMM clustering isn't dominated by 30 constant loci.

## Testing

See the final summary in the PR thread for the full pytest run.

<div><a href="https://cursor.com/agents/bc-d889282a-c84e-4aa6-a1d6-308a8643c8d9"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-d889282a-c84e-4aa6-a1d6-308a8643c8d9"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by csmangum** [comment]
@cursor are these new genes already implemented in the simulation?

**Comment by csmangum** [comment]
@cursor go ahead and do the follow-ups in this pr.
